### ISSUE TITLE
Cycleway quest - Prevent easy removal of an existing oneway:bicycle=no tag

### ISSUE THREAD

**Issue by bxl-forever** [streetcomplete/StreetComplete#3795]
A one-way street, tagged with oneway:bicycle=no
User selects the cycleway quest and must provide the type of cycleway in both directions.
The problem is for the direction against motor traffic.  Option 1 is "none".  Option 2 is "none, but cyclist may use road in both directions".

I have seen many users in my city incorrectly select the first option: "none".

That choice is destructive because it will result in isOnewayNotForCyclists==true and will trigger the removal of "oneway:bicycle=no" from the way.

It is not just a simple mistake, it has nasty consequences because it breaks previously valid bicycle routing, and is particularly hard to spot.  I live in Brussels and almost all one-way streets in my area are open to cyclists in both directions, with just roadsigns but no marks on the ground.

I wish there would be a warning here, something that would say: "This road was up to now set as two-way for cyclists, and you made it strictly one-way, are you sure?"

Or even better, merge those two options to only add cycleway:* tags and never remove an existing "oneway:bicycle=no" tag.  (Make it a separate quest if needed.)

**Comment by FloEdelmann** [comment]
Related: #3778, #533

**Comment by matkoniecz** [comment]
I also though about making "none" option more clear (symbolic green arrow?), at least on ways where it would be a contraflow.

**Comment by westnordost** [comment]
I think a warning dialog would be the best option here.

Though, what I don't understand is: If the road was previously tagged with having no cycleway but it is not a oneway for cyclists, why would a user then **change** this selection? After all, when the quest is asked as a resurvey, the previous tagging is pre-selected.

**Comment by matkoniecz** [comment]
> After all, when the quest is asked as a resurvey, the previous tagging is pre-selected.

In some cases SC is unable to interpret tagging and shows both sides blank.

**Comment by westnordost** [comment]
It would only show the side it cannot interpret as blank.

**Comment by bxl-forever** [comment]
I bet that if the quest is asked as a resurvey, we don’t have this issue.  This is only when SC sees no existing cycleway tag.
Thanks.

**Comment by bxl-forever** [comment]
Oh, sorry, you are right.  Apologies for wasting your time, I must have read the bit of code too fast and indeed, the tag seems to get deleted when ` isOnewayNotForCyclists==false`, not `true`.

If seeing a real example can be helpful, here is one street to illustrate the issue:
https://www.openstreetmap.org/way/35964877/history (version #7)

**Comment by westnordost** [comment]
So, err, can you summarize what exactly is the issue again?

**Comment by westnordost** [comment]
You mean the issue you reported is reproducible with that way?

**Comment by bxl-forever** [comment]
Yes, it is reproducible on the way I provided.

To sum it up.  We start with a way with the following tags
oneway=yes
oneway:bicycle=no
no cycleway tag

In my example, "oneway=yes" and "oneway:bicycle=no" correctly reflect the legal situation.

SC prompts "Is there a cycleway here? What type?"
Imagine there are no marks on the ground at all.
The user will see the available choices and select the first option ("none").  As a consequence, SC will add "cycleway=no" but will also remove "oneway:bicycle=no".

After reading some of the comments above, it seems that either adding a dialog or presenting the options differently would help.

I talked with one user who removed valid contraflows, who explained he was influenced by the bicycle logo on the picture; not seeing any logo painted on the ground convinced him to select "none".

![Screenshot_20220222-111404_Street­Complete](https://user-images.githubusercontent.com/23359667/155219319-760444cc-db65-4385-a187-b04511d2c4ca.png)

**Comment by matkoniecz** [comment]
https://www.openstreetmap.org/way/3987524 is an example - found with https://overpass-turbo.eu/s/1gmW

**Comment by westnordost** [comment]
Okay, so the "none but cyclists may use road in both directions" is not prefilled because StreetComplete does not consider this tagging as complete for that side. Complete for that side would require an additional `cycleway:left=no`. Then, or if the app would interpret a missing `cycleway:left=no` in the case of `oneway:bicycle=no` as complete (enough to display), it'd look like this:

![image](https://user-images.githubusercontent.com/4661658/155301236-9bda54fd-fe36-432e-b16a-f4301471b6fd.png)

I think, displaying a warning dialog in this case would be better.

**Comment by westnordost** [comment]
![image](https://user-images.githubusercontent.com/4661658/158676412-69873202-7c4a-4001-a422-3929d434b313.png)

**Comment by markusstraub** [comment]
Thanks for reporting this issue @bxl-forever. I just spent part of my weekend fixing broken one-ways, i.e. streets where either `cycleway=opposite(_lane|_track)` and/or `oneway:bicycle=no` was removed by users of StreetComplete, in Vienna [1], and it was a bit frustrating :/.

Also thank you for the quick fix @westnordost! However, I would like to reopen this issue because it is only partially fixed in my opinion.

There are at least two ways to tag that a one-way road may be used by cyclists in both directions (that can also be used together)
1. `oneway:bicycle=no`, i.e. the "modern" approach
2. `cycleway[:left|:right]=opposite[_lane|_track]` (the most common variant probably being `cycleway=opposite`), i.e. the oldschool approach

**The new warning dialog is not shown in the second case which is still common (at least here in Vienna).**

**-> please consider a road tagged like this the same as a road with `oneway:bicycle=no`.**

[1] some of the problematic changesets:
- https://www.openstreetmap.org/changeset/117863995
- https://www.openstreetmap.org/changeset/119111558
- https://www.openstreetmap.org/changeset/106726243
- https://www.openstreetmap.org/changeset/105685872
- https://www.openstreetmap.org/changeset/116780450
- https://www.openstreetmap.org/changeset/113457699
- https://www.openstreetmap.org/changeset/115807497 
- https://www.openstreetmap.org/changeset/115805917
- https://www.openstreetmap.org/changeset/115807497
- https://www.openstreetmap.org/changeset/118404534

**Comment by westnordost** [comment]
Are you sure that the warning dialog is not shown in that case? Did you test it?

**Comment by markusstraub** [comment]
Yes I tested it in the current version of the app. Here is a map error introduced by myself a few seconds ago: https://www.openstreetmap.org/changeset/119874197

The street only had `cycleway=opposite_lane`, which is now replaced with `cycleway:both=no`. So according to the data now cycling is only permitted in the direction of the one-way.

The only question asked by the app was if I actually surveyed the area.
